components/org.wso2.bfsi.consent.management.common/src/main/java/org/wso2/bfsi/consent/management/common/config/ConsentManagementConfigParser.java

```diff
@@ -0,0 +1,234 @@
+            throw new ConsentManagementRuntimeException("Error occurred while building configuration from" +
+                    " bfsi-consent-management.xml", e);
+        } finally {
+            try {
+                if (inStream != null) {
+                    inStream.close();
+                }
+            } catch (IOException e) {
+                log.error("Error closing the input stream for bfsi-consent-management.xml", e);
+            }
+        }
+    }
+
+
+    /**
+     * Method to read text configs from xml when root element is given.
+     *
+     * @param serverConfig XML root element object
+     * @param nameStack    stack of config names
+     */
+    private void readChildElements(OMElement serverConfig, Stack<String> nameStack) {
+
+        for (Iterator childElements = serverConfig.getChildElements(); childElements.hasNext(); ) {
+            OMElement element = (OMElement) childElements.next();
+            nameStack.push(element.getLocalName());
+            if (elementHasText(element)) {
+                String key = getKey(nameStack);
+                Object currentObject = configuration.get(key);
+                String value = replaceSystemProperty(element.getText());
+                if (currentObject == null) {
+                    configuration.put(key, value);
+                } else if (currentObject instanceof ArrayList) {
+                    ArrayList<String> list = (ArrayList) currentObject;
+                    if (!list.contains(value)) {
+                        list.add(value);
+                        configuration.put(key, list);
+                    }
+                } else {
+                    if (!value.equals(currentObject)) {
+                        ArrayList<Object> arrayList = new ArrayList<>(2);
+                        arrayList.add(currentObject);
+                        arrayList.add(value);
+                        configuration.put(key, arrayList);
+                    }
+                }
+            }
+            readChildElements(element, nameStack);
+            nameStack.pop();
+        }
+    }
+
+    /**
+     * Method to check whether config element has text value.
+     *
+     * @param element root element as a object
+     * @return availability of text in the config
+     */
+    private boolean elementHasText(OMElement element) {
+
+       return StringUtils.isNotBlank(element.getText());
+    }
+
+    /**
+     * Method to obtain config key from stack.
+     *
+     * @param nameStack Stack of strings with names.
+     * @return key as a String
+     */
+    private String getKey(Stack<String> nameStack) {
+
+        StringBuilder key = new StringBuilder();
+        for (int index = 0; index < nameStack.size(); index++) {
+            String name = nameStack.elementAt(index);
+            key.append(name).append(".");
+        }
+        key.deleteCharAt(key.lastIndexOf("."));
+        return key.toString();
```

Comment:
Couldn't we write this using streams? something like below

```suggestion
        return nameStack.stream().collect(Collectors.joining("."));
```

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/1/commits/2089256398b8b821f4bb347077d8da06d684d829